src/kicker/main.c

```diff
@@ -52,19 +55,19 @@ unsigned time = 0;
 uint8_t execute_cmd(uint8_t, uint8_t);
 
 bool is_kicking() {
-    return pre_kick_cooldown_ || millis_left_ || post_kick_cooldown_ ||
+    return pre_kick_cooldown_ || timer_cnts_left_ || post_kick_cooldown_ ||
            kick_wait;
 }
 
 void kick(uint8_t strength) {
     if (is_kicking()) return;
-    pre_kick_cooldown_ = 5;
+    pre_kick_cooldown_ = 5 * TIMER_TO_MS;
     // minimum of 6 ms, we were breaking kickers with low duty cycles
-    // maximum of 6 + 7 == 13 ms
-    // millis_left_ = (int) ((strength / 255.0) * 6.0) + 7;
-    millis_left_ = 13;  // always full kick speed
-    post_kick_cooldown_ = 5;
-    kick_wait = 2000;
+    float time_cnt_flt = ((strength / 255.0f) * 9.0f + 0.8f) * TIMER_TO_MS;
+    timer_cnts_left_ = (int) (time_cnt_flt + 0.5f);
```

Comment:
Why are we adding this here?

Reply:
Oh whoops. could you add a comment for rounding?

Reply:
yup